dotnet/src/Experimental/Orchestration.Flow/Execution/ReActEngine.cs

```diff
@@ -206,42 +207,37 @@ internal async Task<string> InvokeActionAsync(ReActStep actionStep, string chatI
 
             foreach (var variable in actionContextVariables)
             {
-                contextVariables.Set(variable.Key, variable.Value);
+                contextVariables[variable.Key] = variable.Value;
             }
 
             this._logger?.LogDebug("Invoked {FunctionName}. Result: {Result}", targetFunction.Name, result.GetValue<string>());
 
             return result.GetValue<string>() ?? string.Empty;
         }
-        catch (Exception e) when (!e.IsCriticalException())
+        catch (Exception e) when (!e.IsNonRetryable())
         {
             this._logger?.LogError(e, "Something went wrong in action step: {0}.{1}. Error: {2}", targetFunction.PluginName, targetFunction.Name, e.Message);
             return $"Something went wrong in action step: {targetFunction.PluginName}.{targetFunction.Name}. Error: {e.Message} {e.InnerException?.Message}";
         }
     }
 
-    private ContextVariables CreateActionContextVariables(Dictionary<string, string> actionVariables, Kernel kernel, ContextVariables contextVariables)
+    private KernelArguments CreateActionKernelArguments(Dictionary<string, string> actionVariables, KernelArguments context)
```

Comment:
Recommend you update from the term `context` to `kernelArguments` throughtout the code.

Not a "must do" for this PR.

Reply:
Thanks, will try to sort out this in next PR.

In general I'd like to keep `kernelArguments` and `context` separate to differentiate the arguments passed from the call context and the context arguments for the flow session.

A real-life example is that user's AAD token will be available in the `kernelArguments` passed to orchestrator, in transient way, while the state variables would be available in `context` which are synchronized to state persistence layer.